installer/pkg/config-generator/tls.go

```diff
@@ -1,56 +1,174 @@
+		return fmt.Errorf("failed to generate kube CAs: %v", err)
+	}
+	// generate aggregator CA
+	if err := generateCACerts(clusterDir, caKey, caCert, aggregatorCAKeyPath, aggregatorCACertPath, "aggregator", "bootkube", x509.KeyUsageKeyEncipherment|x509.KeyUsageDigitalSignature|x509.KeyUsageCertSign); err != nil {
+		return fmt.Errorf("failed to generate aggregator CAs: %v", err)
+	}
+	// generate service-serving CA
+	if err := generateCACerts(clusterDir, caKey, caCert, serviceServiceCAKeyPath, serviceServiceCACertPath, "service-serving", "bootkube", x509.KeyUsageKeyEncipherment|x509.KeyUsageDigitalSignature|x509.KeyUsageCertSign); err != nil {
+		return fmt.Errorf("failed to generate kube CAs: %v", err)
+	}
+	return nil
+}
+
+// generatePrivateKey generates and writes the private key to disk
+func generatePrivateKey(clusterDir string, path string) (*rsa.PrivateKey, error) {
+	fileTargetPath := filepath.Join(clusterDir, path)
+	key, err := tls.PrivateKey()
+	if err != nil {
+		return nil, fmt.Errorf("error writing private key: %v", err)
+	}
+	if err := writeFile(fileTargetPath, tls.PrivateKeyToPem(key)); err != nil {
+		return nil, err
+	}
+	return key, nil
+}
+
+// generateRootCerts creates the rootCAKey and rootCACert
+func generateRootCerts(clusterDir string) (cert *x509.Certificate, key *rsa.PrivateKey, err error) {
+	// generate key and certificate
+	caKey, err := generatePrivateKey(clusterDir, rootCAKeyPath)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to generate private key: %v", err)
+	}
+	caCert, err := rootCA(clusterDir, caKey)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to create a certificate: %v", err)
+	}
+	return caCert, caKey, nil
+}
+
+// getCertFiles copies the given cert/key files into the generated folder and returns their contents
+func getCertFiles(clusterDir string, certPath string, keyPath string) (*x509.Certificate, *rsa.PrivateKey, error) {
+	keyDst := filepath.Join(clusterDir, rootCAKeyPath)
+	if err := copyFile(keyPath, keyDst); err != nil {
+		return nil, nil, fmt.Errorf("failed to write file: %v", err)
+	}
+
+	certDst := filepath.Join(clusterDir, rootCACertPath)
+	if err := copyFile(certPath, certDst); err != nil {
+		return nil, nil, fmt.Errorf("failed to write file: %v", err)
+	}
+	// content validation occurs in pkg/config/validate.go
+	// if it fails here, something went wrong
+	certData, err := ioutil.ReadFile(certPath)
+	if err != nil {
+		panic(err)
+	}
+	certPem, _ := pem.Decode([]byte(string(certData)))
+	keyData, err := ioutil.ReadFile(keyPath)
+	if err != nil {
+		panic(err)
+	}
+	keyPem, _ := pem.Decode([]byte(string(keyData)))
+	key, err := x509.ParsePKCS1PrivateKey(keyPem.Bytes)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to process private key: %v", err)
+	}
+	cert, err := x509.ParseCertificates(certPem.Bytes)
```

Comment:
nit: since this is a slice of `*x509.Certificate` the variable should be pluralized to `certs`. not a blocker